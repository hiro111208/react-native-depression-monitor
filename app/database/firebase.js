//import firebase from "firebase/app";
import "firebase/firestore";
import firebase from 'firebase'
require('firebase/auth')

const firebaseConfig = {
  apiKey: "AIzaSyDwOUkdxC7kNNILoFTSx05PTUxRci_kcaw",
  authDomain: "fireship-tutorial-646fc.firebaseapp.com",
  projectId: "fireship-tutorial-646fc",
  storageBucket: "fireship-tutorial-646fc.appspot.com",
  messagingSenderId: "289064122528",
  appId: "1:289064122528:web:ee39abfa2183c8499071c5",
  measurementId: "G-D35RDH9V0E",
};

firebase.initializeApp(firebaseConfig);

export default firebase;
