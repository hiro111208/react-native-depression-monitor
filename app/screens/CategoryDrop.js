import React from "react";
import { StyleSheet, View, Text, TouchableOpacity } from "react-native";
import * as indexStyles from "../config/indexStyles";
import firebase from "../database/firebase";

/*
 * Option for the user to drop a category
 */
const CategoryDrop = ({ route, navigation }) => {
    // Stores user data passed from the previous screen
    const userProgress = route.params.user;

    // Reference to the path of the user's data
    const ref = firebase
        .firestore()
        .collection("users")
        .doc(firebase.auth().currentUser.uid);

    // Drops the category for the user or assigns them to control data set
    function chooseOption(chosenCategory) {
        var num = Math.random();
        if (num <= 0.5) {
            chosenCategory = "CONTROL";
        }
        userProgress.categoryDropped = chosenCategory;

        //Update the database with new category
        ref.set(
            {
                categoryDropped: userProgress.categoryDropped,
                lastActive: new Date(),
            },
            { merge: true }
        )
            .then(() => {
                console.log(
                    "Category successfully dropped" +
                        userProgress.categoryDropped
                );
                navigation.navigate("PatientDashboard");
            })
            .catch((error) => {
                console.error("Error writing document: ", error);
            });
    }

    return (
        <View style={styles.container}>
            <View style={[styles.top, indexStyles.centering]}>
                <Text style={styles.text} testID={"INSTRUCTIONS"}>
                    Drop the subject you are least interested in.
                </Text>
            </View>

            <View style={styles.bottom}>
                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("SOCIAL")}
                    testID={"SOCIAL_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Social</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("ACADEMIC")}
                    testID={"ACADEMIC_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Academic</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("HEALTH")}
                    testID={"MOOD_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Mood</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("HEALTH")}
                    testID={"HEALTH_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Health</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("HOBBIES")}
                    testID={"HOBBIES_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Hobbies</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("FAMILY")}
                    testID={"FAMILY_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Family</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("WORK")}
                    testID={"WORK_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Work</Text>
                    </View>
                </TouchableOpacity>

                <TouchableOpacity
                    style={styles.bottomItem}
                    onPress={() => chooseOption("RELATIONSHIP")}
                    testID={"RELATIONSHIP_BUTTON"}
                >
                    <View
                        style={[
                            styles.bottomItemInner,
                            indexStyles.shadowEffect,
                            indexStyles.centering,
                        ]}
                    >
                        <Text style={styles.text}>Relationship</Text>
                    </View>
                </TouchableOpacity>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#ffd394",
    },
    text: {
        color: "black",
        fontSize: 20,
    },
    top: {
        height: "10%",
    },
    bottom: {
        height: "90%",
        flexDirection: "row",
        flexWrap: "wrap",
        padding: 5,
    },
    bottomItem: {
        width: "50%",
        height: "20%",
        padding: 10,
    },
    bottomItemInner: {
        flex: 1,
        backgroundColor: "white",
        borderRadius: 10,
    },
});

export default CategoryDrop;
