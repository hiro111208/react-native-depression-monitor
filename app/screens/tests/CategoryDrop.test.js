import React from "react";
import { shallow } from "enzyme";
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import CategoryDrop from "../CategoryDrop";
import * as firebase from "firebase";

beforeAll(() => {
    jest.useFakeTimers();
});

jest.spyOn(firebase, "auth").mockImplementation(() => {
    return {
        currentUser: { uid: "123456" },
    }
})

describe("Testing CategoryDrop.js rendering when the user has not dropped a category", () => {
    
    it("CategoryDrop shallow renders correctly", () => {
        const wrapper = shallow(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.length).toBe(1);
    });

    it("CategoryDrop renders correctly", () => {
        const tree = renderer.create(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Renders instructions", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("INSTRUCTIONS")).toBeTruthy();
    });
    
    it("Renders social button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("SOCIAL_BUTTON")).toBeTruthy();
    });

    it("Renders acadmeic button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("ACADEMIC_BUTTON")).toBeTruthy();
    });  

    it("Renders mood button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("MOOD_BUTTON")).toBeTruthy();
    });  

    it("Renders health button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("HEALTH_BUTTON")).toBeTruthy();
    });  

    it("Renders hobbies button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("HOBBIES_BUTTON")).toBeTruthy();
    });  

    it("Renders family button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("FAMILY_BUTTON")).toBeTruthy();
    });  

    it("Renders work button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("WORK_BUTTON")).toBeTruthy();
    });

    it("Renders relationship button", () => {
        const wrapper = render(<CategoryDrop route={{params: {user: {categoryDropped: "NONE"}}}}/>);
        expect(wrapper.queryByTestId("RELATIONSHIP_BUTTON")).toBeTruthy();
    });  

});

describe("Selecting a category navigates to PatientDashboard", () => {

    it("Selecting a category navigates to PatientDashboard", () => {
        const navigation = { navigate: jest.fn() };
        const wrapper = render(<CategoryDrop route={{ params: { user: { categoryDropped: "NONE" }}}} navigation={ navigation } />);
        const button = wrapper.getByTestId("HEALTH_BUTTON");
        expect(navigation.navigate).not.toHaveBeenCalledWith("PatientDashboard");
        fireEvent(button, "press");
    });

    it("chooseOption not called until button pressed", () => {
        const chooseOption = jest.fn(() => true);
        const wrapper = render(<CategoryDrop route={{ params: { user: { categoryDropped: "NONE" }}}} />);
        const button = wrapper.getByTestId("HEALTH_BUTTON");
        expect(chooseOption).not.toHaveBeenCalledWith("HEALTH");
        fireEvent(button, "press");
    });
        
});

// describe("Testing CategoryDrop.js buttons store user data correctly", () => {
// });

afterAll(cleanup);