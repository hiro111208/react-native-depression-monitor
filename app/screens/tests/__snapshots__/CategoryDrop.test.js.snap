// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CategoryDrop.js rendering when the user has not dropped a category CategoryDrop renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffd394",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": "10%",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
        }
      }
      testID="INSTRUCTIONS"
    >
      Drop the subject you are least interested in.
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": "90%",
        "padding": 5,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="SOCIAL_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Social
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="ACADEMIC_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Academic
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="MOOD_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Mood
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="HEALTH_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Health
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="HOBBIES_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Hobbies
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="FAMILY_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Family
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="WORK_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Work
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "20%",
          "opacity": 1,
          "padding": 10,
          "width": "50%",
        }
      }
      testID="RELATIONSHIP_BUTTON"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "flex": 1,
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
            }
          }
        >
          Relationship
        </Text>
      </View>
    </View>
  </View>
</View>
`;
